### PR TITLE
Added better error handling when a DB driver is wrong defined (fix #10)

### LINKED ISSUE #10: Misleading error on loading plugin if a DB driver defined wrong in config
```json
 "databases": [
        {
            "name": "dbtest",
            "driver": "mtsql",   <----------------------------mistype or unsupported driver
            "driver_option": {
                "host": "dbhost.local",
                "port": "5432",
                "username": "psql",
                "password": "",
                "dbname": "db_test"
            },
            "dbqueries": [
                {
                    "query": "task_count"
                }
            ]
        }
    ],
```
```sh
$ snapctl plugin load plugins_use/snap-plugin-collector-dbi
Error loading plugin:
unexpected EOF
```

I think some json values may be checked before using them, and the error could be like this:
```sh
Error loading plugin:
GetMetricTypes call error : Driver "mtsql" not supported; Currently available: mysql, postgresql, etc.
```

when using proper "mysql" driver, the error is OK:
```
Error loading plugin:
GetMetricTypes call error : Cannot open any of defined database
```

### PR DESCRIPTION
**Pull request by daria-lewandowska**
Fixes #10 
Summary of changes:
-better error handling

Testing done:
-integration test

### CODE REVIEW

File: dbi/mock/mockData.go [hunk 1]
```diff
@@ -150,4 +150,73 @@ var (
 		        }
 		    ]
 		}`)
+	// FileContCorr is a mocked content of incorrect setfile
+	FileContCorr = []byte(`{
```
Comment (marcin-krolik) [inline dbi/mock/mockData.go]: Consider moving `json` content as separate file and read it

File: dbi/dbi.go [hunk 2]
```diff
@@ -153,7 +150,6 @@ func (dbiPlg *DbiPlugin) getMetrics() (map[string]interface{}, error) {
 	metrics := map[string]interface{}{}
 
 	err := openDBs(dbiPlg.databases)
-	defer closeDBs(dbiPlg.databases)
```
Comment (marcin-krolik) [inline dbi/dbi.go]: What is the reason behind removing `defer` statement and moving `closeDBs()` to the end of function?

File: dbi/connect.go [hunk 3]
```diff
@@ -59,9 +58,8 @@ func openDB(db *dtype.Database) error {
 			db.Username, db.Password, db.Host, db.Port, db.DBName)
 
 	default:
-		return fmt.Errorf("SQL Driver %s is not supported", db.Driver)
+		return errors.New("SQL Driver " + db.Driver + " is not supported")
```
Comment (marcin-krolik) [inline dbi/connect.go]: Consider using `fmt.Sprintf()` instead string concat

File: dbi/dbi.go (hunk 2 above)
Comment (daria-lewandowska) [inline dbi/dbi.go]: If there was a mislead in drivers name the connection wasn't opened, so we try to close connection which wasn't opened, this causes "unexpected EOF" error. Now we are closing only opened connections.

Comment (marcin-krolik) [inline dbi/dbi.go]: :+1:

**Comment by marcin-krolik** [comment]
There is still incorrectly handled error in `closeDBs()`

**Comment by daria-lewandowska** [comment]
I'm not sure how should I handle this error in closeDBs(), 
you think it will be ok to do: 
`return fmt.Errorf("Cannot close connection to database %+v, err=%+v\n", dbs[i].DBName)`
instead of: 
`fmt.Fprintf(os.Stdout, "Cannot close connection to database %+v, err=%+v\n", dbs[i].DBName, err)`

File: dbi/mock/mockData.go (hunk 1 above)
Comment (daria-lewandowska) [inline dbi/mock/mockData.go]: done

File: dbi/connect.go (hunk 3 above)
Comment (daria-lewandowska) [inline dbi/connect.go]: done

**Comment by marcin-krolik** [comment]
@daria-lewandowska yes, I'd suggest to add `error` as returned type to `closeDBS()` function, but would not return early from loop. I think you should try to close all existing DB connections, and if any fails, return an error

**Comment by marcin-krolik** [comment]
LGTM

**Pull request merged**